### DISCUSSION TITLE
API /icons is marked deprecated

### DISCUSSION THREAD

**Discussion by ktrue** [weather-gov/api#557] [General]
The current specification at https://www.weather.gov/documentation/services-web-api shows that 

/icons
/icons/{set}/{timeOfDay}/{first}
/icons/{set}/{timeOfDay}/{first}/{second}

are all marked as deprecated.  However a gridpoint request like 
https://api.weather.gov/gridpoints/MTR/94,80/forecast

still returns data referencing the /icons/ API.

```
        "periods": [
            {
                "number": 1,
                "name": "This Afternoon",
                "startTime": "2022-06-18T14:00:00-07:00",
                "endTime": "2022-06-18T18:00:00-07:00",
                "isDaytime": true,
                "temperature": 72,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "14 mph",
                "windDirection": "NW",
                "icon": "https://api.weather.gov/icons/land/day/skc?size=medium",
                "shortForecast": "Sunny",
                "detailedForecast": "Sunny, with a high near 72. Northwest wind around 14 mph, with gusts as high as 21 mph."
            },
            {
                "number": 2,
                "name": "Tonight",
                "startTime": "2022-06-18T18:00:00-07:00",
                "endTime": "2022-06-19T06:00:00-07:00",
                "isDaytime": false,
                "temperature": 49,
                "temperatureUnit": "F",
                "temperatureTrend": null,
                "windSpeed": "1 to 13 mph",
                "windDirection": "WSW",
                "icon": "https://api.weather.gov/icons/land/night/few?size=medium",
                "shortForecast": "Mostly Clear",
                "detailedForecast": "Mostly clear, with a low around 49. West southwest wind 1 to 13 mph, with gusts as high as 21 mph."
            },
```
Note the "icon": values.

What is the "icon": value intended to be replaced with if the /icons API is discontinued, and when might this happen?

**Comment by scadergit** [comment]
A more generic value that is not a url so it can be mapped to any graphic representation, including the current forecast icons. 

As a user of the API for other weather apps, I wish it was yesterday. Alas, there is no timeline yet. Hopefully, when the API is updated with pop and other information, rather than a direct copy of the current forecast application json.

**Comment by ktrue** [comment]
Hopefully, the new values will reflect the prior icon names so an easy mapping can be done to existing images.

I see that weather.gov uses forecast.weather.gov/newimages/... for the images, including the use of DualImage.php to dynamically generate the split 6hr:6hr images when needed.   Also the NOAA Weather Homepage widget uses [forecast.weather.gov/MapClick.php?...type=json](https://forecast.weather.gov/MapClick.php?lat=37.27105000000006&lon=-122.01400999999998&unit=0&lg=english&FcstType=json)
to acquire data.  Should we be using this instead of the api.weather.gov for forecasts until the API is fully settled?
It does include icon links and is quite simple to parse with one query to get all the data for the point lat/long.

**Comment by scadergit** [comment]
The API is the future, but it's your call when it's complete enough to transition (aside from when mapclick json sunsets, which has no timeline yet).

**Comment by justinferrell** [comment]
Has this moved at all? I'm building against the API and trying to get a finite list of possible conditions so we can provide our own icons. The system icons made sense as keys but if they're going away, I'm not sure what would work.

**Comment by scadergit** [comment]
It won't officially go away without much ado and a formal replacement, as
use the same for official NWS applications.

**Comment by laurabennett** [comment]
Any updates? We are also looking for a distinct set of values that we can map to custom icons. The values being returned in the deprecated icon field have recently changed and no longer match the previous values exactly so we are getting errors.

**Reply by scottshaffer** [reply]
I have the same question - the json for icons is now quite odd to me and I don't know how to process it.  For example, a recent one for a given period says "tsra_hi,20/tsra_hi,50" - does anybody know what that means?   The text description matches part of that (Chance of precipitation is 50%), but not all of it.  I thought perhaps it was showing morning vs evening but the period is already indicating that and the 2 periods have different values.  Note this is in the gridpoint forecast.

**Reply by ktrue** [reply]
It's the 6-hour split forecast.  First part is using tsra_hi icon with 20% PoP, second 6-hour period is tsra_hi icon with 50% PoP likely.
They are displayed as 'split' icons by the DualImage.php script on forecast.weather.gov like this:

https://forecast.weather.gov/DualImage.php?i=hi_ntsra&j=hi_ntsra&ip=20&jp=50

My PHP forecast script (https://github.com/ktrue/NWS-forecast) has my version of the DualImage.php script to re-create the dual icon with PHP GD library.

**Reply by scadergit** [reply]
@laurabennett the icon path fix should be deployed soon; technical issues beyond the API and critical weather have delayed the patch.

@scottshaffer each forecast day has two, 12-hour periods (day/night), and each period can have one or two icons based changing weather over the period. In your example, it changed from 20% of thunderstorms to 50% of thunderstorms in the same period. If there is no `/` in the icon path then the weather is constant and only one icon suggestion is provided.

**Reply by tmabell** [reply]
@ktrue I noticed that as of yesterday the icon path has been corrected.
